contrib/devtools/check-rpc-mappings.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python3
+'''
+Check RPC argument consistency.
+'''
+import sys, os, re
+from collections import defaultdict
+
+# Source files (relative to root) to scan for dispatch tables
+SOURCES = [
+"src/rpc/server.cpp",
+"src/rpc/blockchain.cpp",
+"src/rpc/mining.cpp",
+"src/rpc/misc.cpp",
+"src/rpc/net.cpp",
+"src/rpc/rawtransaction.cpp",
+"src/wallet/rpcwallet.cpp",
+]
+# Source file (relative to root) containing conversion mapping
+SOURCE_CLIENT = 'src/rpc/client.cpp'
+# Argument names that should be ignored in consistency checks
+IGNORE_DUMMY_ARGS = {'dummy','arg0','arg1','arg2','arg3','arg4','arg5','arg6','arg7','arg8','arg9'}
+
+def parse_string(s):
+    assert(s[0] == '"')
+    assert(s[-1] == '"')
+    return s[1:-1]
+
+def make_string(s):
+    return '"%s"' % s
+
+class RPCCommand:
+    def __init__(self, name, args):
+        self.name = name
+        self.args = args
+
+class RPCArgument:
+    def __init__(self, names, idx):
+        self.names = names
+        self.idx = idx
+
+def process_commands(fname):
```

Comment:
I think it'd simplify implementation in `main()` if `process_commands()` and `process_mappings()` both returned a set of tuples `(command, index, argname)`.